src/main/scala/mesosphere/util/state/FrameworkIdUtil.scala

```diff
@@ -0,0 +1,58 @@
+package mesosphere.util.state
+
+import org.apache.mesos.Protos.{ FrameworkInfo => FrameworkInfoProto, FrameworkID => FrameworkIDProto }
+import com.google.protobuf.InvalidProtocolBufferException
+import java.util.logging.{ Level, Logger }
+import scala.concurrent.duration.Duration
+import scala.util.{ Failure, Success }
+import scala.concurrent.{ Future, Await, ExecutionContext }
+
+/**
+  * Utility class for keeping track of a framework ID in Mesos state.
+  *
+  * @param state State implementation
+  * @param key The key to store the framework ID under
+  */
+
+class FrameworkIdUtil(val state: PersistentStore, val key: String = "frameworkId") {
```

Comment:
Somehow I do not find the deleted FrameworkIdUtil in the changes, strange. 

Why do you use `val` here? And wouldn't it be to use `MarathonStore` instead of `PersistentStore`?